http://mail.madler.net/pipermail/zlib-devel_madler.net/2011-June/002583.html

[Zlib-devel] Possible typo in zlib.h

Mark Adler madler at madler.net

*** a/zlib.h    2010-04-19 21:12:48.000000000 -0700
--- b/zlib.h  2011-02-26 21:23:30.000000000 -0800
***************
*** 1572,1584 ****
  #endif
  
  #if !defined(ZLIB_INTERNAL) && _FILE_OFFSET_BITS-0 == 64 && _LFS64_LARGEFILE-0
! #  define gzopen gzopen64
! #  define gzseek gzseek64
! #  define gztell gztell64
! #  define gzoffset gzoffset64
! #  define adler32_combine adler32_combine64
! #  define crc32_combine crc32_combine64
! #  ifdef _LARGEFILE64_SOURCE
       ZEXTERN gzFile ZEXPORT gzopen64 OF((const char *, const char *));
       ZEXTERN z_off_t ZEXPORT gzseek64 OF((gzFile, z_off_t, int));
       ZEXTERN z_off_t ZEXPORT gztell64 OF((gzFile));
--- 1580,1601 ----
  #endif
  
  #if !defined(ZLIB_INTERNAL) && _FILE_OFFSET_BITS-0 == 64 && _LFS64_LARGEFILE-0
! #  ifdef Z_PREFIX_SET
! #    define z_gzopen z_gzopen64
! #    define z_gzseek z_gzseek64
! #    define z_gztell z_gztell64
! #    define z_gzoffset z_gzoffset64
! #    define z_adler32_combine z_adler32_combine64
! #    define z_crc32_combine z_crc32_combine64
! #  else
! #    define gzopen gzopen64
! #    define gzseek gzseek64
! #    define gztell gztell64
! #    define gzoffset gzoffset64
! #    define adler32_combine adler32_combine64
! #    define crc32_combine crc32_combine64
! #  endif
! #  ifndef _LARGEFILE64_SOURCE
       ZEXTERN gzFile ZEXPORT gzopen64 OF((const char *, const char *));
       ZEXTERN z_off_t ZEXPORT gzseek64 OF((gzFile, z_off_t, int));
       ZEXTERN z_off_t ZEXPORT gztell64 OF((gzFile));
*** a/zconf.h.in    2010-04-18 10:58:06.000000000 -0700
--- b/zconf.h.in  2010-07-18 09:05:49.000000000 -0700
***************
*** 374,380 ****
  #  undef _LARGEFILE64_SOURCE
  #endif
  
! #if defined(Z_HAVE_UNISTD_H) || defined(_LARGEFILE64_SOURCE)
  #  include <unistd.h>       /* for SEEK_* and off_t */
  #  ifdef VMS
  #    include <unixio.h>     /* for off_t */
--- 375,385 ----
  #  undef _LARGEFILE64_SOURCE
  #endif
  
! #if defined(_LARGEFILE64_SOURCE) && _LFS64_LARGEFILE-0
! #  define Z_LARGE
! #endif
! 
! #if defined(Z_HAVE_UNISTD_H) || defined(Z_LARGE)
  #  include <unistd.h>       /* for SEEK_* and off_t */
  #  ifdef VMS
  #    include <unixio.h>     /* for off_t */
